### ISSUE TITLE
unable to make `u-boot` and so `hwpack` too

### ISSUE THREAD

**Issue by WebReflection** [linux-sunxi/sunxi-bsp#11]
it looks like latest `u-boot` does not want `boardname_config` but just `boardname`.

At least this is the change I've made in `Makefile` modifying `$(UBOOT_CONFIG)` dropping the `_config` part.

This was necessary after I've manually updated each submodule to the latest.

**Comment by arokux** [comment]
On Sat, Sep 28, 2013 at 10:44 PM, Andrea Giammarchi <
notifications@github.com> wrote:

> it looks like latest u-boot does not want boardname_config but just
> boardname.
> 
> At least this is the change I've made in Makefile modifying
> $(UBOOT_CONFIG) dropping the _config part.
> 
> This was necessary after I've manually updated each submodule to the
> latest.
> 
> Is it different from what is described here?

https://github.com/linux-sunxi/u-boot-sunxi/wiki

**Comment by hno** [comment]
lör 2013-09-28 klockan 13:44 -0700 skrev Andrea Giammarchi:

> it looks like latest u-boot does not want boardname_config but just
> boardname.

boardname_config is fine, this configures u-boot. You then need to run
make again to compile.

without _config it both configures and compiles in one go.

Regards
Henrik

**Comment by WebReflection** [comment]
might be ok but it does not work ... 

```
make u-boot
mkdir -p /home/webreflection/sunxi-bsp/build/pcduino-u-boot
make -C u-boot-sunxi pcduino_config O=/home/webreflection/sunxi-bsp/build/pcduino-u-boot CROSS_COMPILE=arm-linux-gnueabihf- -j4
make[1]: Entering directory `/home/webreflection/sunxi-bsp/u-boot-sunxi'
make: *** No rule to make target `pcduino_config'.  Stop.
make[1]: *** [pcduino_config] Error 1
make[1]: Leaving directory `/home/webreflection/sunxi-bsp/u-boot-sunxi'
make: *** [/home/webreflection/sunxi-bsp/build/pcduino-u-boot/include/config.h] Error 2
```

and I can try with any board in the list, cubieboard and others ... same thing.

u-boot is the latest master

**Comment by WebReflection** [comment]
to make it clear ... if I enter in the `u-boot-sunxi` folder manually and write `make pcduino CROSS_COMPILE=arm-linux-gnueabihf-` it builds without problems.

If I do

```
./configure pcduino
make
```

from `sunxi-bsp` folder I am unable to build `u-boot` because of the config.

This happens only after updating `u-boot` to latest master (sunxi branch) instead of using the one specified in this repo since those are quite outdated commits ... right?

So, long story short, it's not possible anymore to build via this utility branch because `u-boot` does not work as expected through configuration steps performed by `./configure` in this repo.

I hope this is better/cleaner now.

**Comment by techn** [comment]
@WebReflection, you seem to be right. *_config doesn't work anymore. I'll try to check this tomorrow. BSP needs update anyway.

**Comment by WebReflection** [comment]
thank you @techn ... bear in mind that my proposed change does not seem to work at all. It looks like everything is OK but then it does not boot anymore through `scripts/sunxi-media-create.sh`

Hope this will be fixed since this utility "all in one" repo is pretty awesome for quick tests/changes prototypes.

Thanks again

**Comment by hno** [comment]
The u-boot board name is pcDuino with capital D.

Regards
Henrik

lör 2013-09-28 klockan 15:48 -0700 skrev Andrea Giammarchi:

**Comment by WebReflection** [comment]
@hno are you trolling on purpose? ... I cannot `configure` using non existent options plus @tech already confirmed the problem ... you cannot `./configure pcDuino` from this repo also because the fex script is lowercase too

**Comment by hno** [comment]
Not trolling, only stating what is the core of the problem.

sunxi u-boot now accepts all lowercase names in _config targets as well. But as I said before the u-boot board name is really pcDuino. It's already been rejected upstream to make u-boot board names case insensitive in make.

**Comment by WebReflection** [comment]
it does not work in any case, no matter how I call it, because the configure script does not have a pcDuino in the list so it keeps giving me the list without going further on.

You really should try to read/understand wht is the problem here which is **not** about u-boot, as I've said, which compiles without problems using `pcduino` as board so I have no idea what your hint could solve for this bsp folder problem, sorry,

**Comment by hno** [comment]
Just update your u-boot tree and it should work.

**Comment by WebReflection** [comment]
I've seen a change there, thank you. But still it does not work. Actually, it's worst than before due `awk: line 1: missing ) near }` error.

I am trying to figure out where is the error in the new `Makefile` ... no luck so far but thanks for trying helping.

**Comment by WebReflection** [comment]
[the error and the way I've solved](https://github.com/linux-sunxi/u-boot-sunxi/commit/468752b06c67aafba5004a4a00dd0d4b8e1cc977#commitcomment-4206835)

that said, current status after a `make update`:

```
root@webreflection-ASUS:/home/webreflection/sunxi-bsp# make
mkdir -p /home/webreflection/sunxi-bsp/build/pcduino-u-boot
make -C u-boot-sunxi pcduino_config O=/home/webreflection/sunxi-bsp/build/pcduino-u-boot CROSS_COMPILE=arm-linux-gnueabihf- -j4
make[1]: Entering directory `/home/webreflection/sunxi-bsp/u-boot-sunxi'
make[1]: Leaving directory `/home/webreflection/sunxi-bsp/u-boot-sunxi'
make[1]: Entering directory `/home/webreflection/sunxi-bsp/u-boot-sunxi'
make: *** No rule to make target `pcduino_config'.  Stop.
make[1]: *** [pcduino_config] Error 1
make[1]: Leaving directory `/home/webreflection/sunxi-bsp/u-boot-sunxi'
make: *** [/home/webreflection/sunxi-bsp/build/pcduino-u-boot/include/config.h] Error 2
```

**Comment by WebReflection** [comment]
for github reference sake, [the awk error fix](https://github.com/linux-sunxi/u-boot-sunxi/pull/51)

**Comment by amery** [comment]
beside the _config thing, the main issue is the inconsistency between the name in sunxi-boards and u-boot-sunxi's boards.cfg. for the BSP to work both MUST match.

**Comment by WebReflection** [comment]
I understand that but it was actually an infortunate choice to go _case sensitive_ which does not really help much for any particular purpose. `pcduino` and `pcDuino` ... who really cares, IMO, but now all folders, fex files, and others need to be aligned with `u-boot` and let the mess start: is it `Cubieboard`, `cubieBoard`, or just `cubieboard` ? is it `MarsBoard` or others? ... bad choice guys, bad choice :-/

**Comment by techn** [comment]
Seems that %_config target fails because mkconfig.

make[1]: Entering directory `/home/techn/sunxi-bsp/u-boot-sunxi'
rm -f /home/techn/sunxi-bsp/build/a13_mid-u-boot/include/config.h /home/techn/sunxi-bsp/build/a13_mid-u-boot/include/config.mk \
        /home/techn/sunxi-bsp/build/a13_mid-u-boot/board/*/config.tmp /home/techn/sunxi-bsp/build/a13_mid-u-boot/board/*/*/config.tmp \
        /home/techn/sunxi-bsp/build/a13_mid-u-boot/include/autoconf.mk /home/techn/sunxi-bsp/build/a13_mid-u-boot/include/autoconf.mk.dep \
        /home/techn/sunxi-bsp/build/a13_mid-u-boot/include/spl-autoconf.mk \
        /home/techn/sunxi-bsp/build/a13_mid-u-boot/include/tpl-autoconf.mk
/home/techn/sunxi-bsp/u-boot-sunxi/mkconfig -A a13_mid
make: *** No rule to make target`a13_mid_config'.  Stop.

**Comment by techn** [comment]
This seems to help.. but not sure how this should be fixed. To be case sensitive requires a lot of work.

 if [ ( $# -eq 2 ) -a ( "$1" = "-A" ) ] ; then
    # Automatic mode
-   line=`awk '($0 !~ /^#/ && $7 ~ /^'"$2"'$/) { print $1, $2, $3, $4, $5, $6, $7, $8 }' boards.cfg`
-   line=`awk '($0 !~ /^#/ && tolower($7) ~ /^'"$2"'$/) { print $1, $2, $3, $4, $5, $6, $7, $8 }' boards.cfg`
  if [ -z "$line" ] ; then

**Comment by WebReflection** [comment]
is anyone taking care of this problem ? also … how is `u-boot` working since all `FEL` files are still lowercase? Thanks for any info/update on this. If there's anything I could do let me know.

**Comment by WebReflection** [comment]
OK, `Marsboard_A10` renamed as `marsboard_a10` worked with the config, using `./configure Marsboard_A10` upper case regardless … but `pcDuino` and `Cubieboard` are still broken in this repo :-(

**Comment by hno** [comment]
mån 2013-10-14 klockan 22:20 -0700 skrev Andrea Giammarchi:

> OK, Marsboard_A10 renamed as marsboard_a10 worked with the config,
> using ./configure Marsboard_A10 upper case regardless … but pcDuino
> and Cubieboard are still broken in this repo :-(

Should work now.

Regards
Henrik

**Comment by WebReflection** [comment]
Hi Henrik,
  after a `git submodule foreach git fetch` from a fresh clone performed yesterday, I can `./configure Marsboard_A10 && make` without problems but I still cannot `./configure cubieboard && make` or `./configure pcduino && make`, same identical error I had at the beginning.

I've seen the `Marsboard_A10` file has been renamed into `marsboard_a10` but others (cubie, pcduino) where already lowercase but still nothing changed.

Regards

**Comment by WebReflection** [comment]
Wait, I wasn't updating properly. So, doing this:

```
# git clone https://github.com/linux-sunxi/sunxi-bsp
cd sunxi-bsp
git submodule init
git submodule update
git submodule foreach git fetch origin HEAD
```

I can finally build again `pcduino`, `cubieboard`, and `marsboard_a10` all lowercase without problems.

@hno do you mind telling me what happened to the "must be case-sensitive" story ? Has this been solved in `u-boot` as change of direction or what?

Thank you in any case for helping out.

Best Regards
